server/src/main/java/com/google/speech/tools/voxetta/servlets/UtteranceUploadServlet.java

```diff
@@ -12,80 +12,52 @@
- 
-  private DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
-  private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
- 
+
+  private DatastoreUtteranceService service = new DatastoreUtteranceService(); 
+  private String audio;
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
- 
-    // Get the URL of the audio file that has been uploaded to Blobstore
-    String audioKey = getUploadedFileKey(request, "audio");
-    
-    Entity utteranceEntity = new Entity("Utterance");
-    utteranceEntity.setProperty("audioKey", audioKey);
-    utteranceEntity.setProperty("userId", "FILLER");
-    utteranceEntity.setProperty("promptId", "FILLER");
-    utteranceEntity.setProperty("device", "FILLER");
-    utteranceEntity.setProperty("age", 100);
-    utteranceEntity.setProperty("gender", "FILLER");
-    
-    // Store the Utterance Entity in Datastore 
-    datastoreService.put(utteranceEntity);
- 
-    // Redirect back to the main recording page
-    response.sendRedirect("/index.html");
-  }
- 
-  // Returns the BlobKey of an uploaded file, or null if a file was not uploaded
-  private String getUploadedFileKey(HttpServletRequest request, String formInputElementName) {
-    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
-    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+    response.setContentType("application/json");
 
-    // Return null if an audio file was not uploaded
-    if (blobKeys == null || blobKeys.isEmpty()) {
-      return null; 
+    // Get the BlobKey of the audio file that has been uploaded to Blobstore
+    try {
+      audio = service.getAudio(request);
+    } catch (IllegalStateException impossible) {
+      throw new AssertionError();
     }
- 
-    // Return the BlobKey of the uploaded file
-    return blobKeys.get(0).getKeyString();
-  }
 
-  /** 
-  * Allow the servlet's Datastore service to be set for mocking purposes
-  */
-  public void setDatastoreService(DatastoreService inputService) {
-      datastoreService = inputService; 
+    // Create and save Utterance to Datastore
+    Utterance utterance = new Utterance(audio, "FILLER", "FILLER", "FILLER", 100, "FILLER");
+    try {
+      service.saveUtterance(utterance);
+      response.getWriter().println("{ \"success\": true }"); 
+    } catch (DatastoreFailureException e) {
+      response.getWriter().println("{ \"success\": false, \"error\": \"Error: Failed to upload Utterance to Datastore.\" }");
+    }
   }
 
   /** 
-  * Allow the servlet's Blobstore service to be set for mocking purposes
-  */
-  public void setBlobstoreService(BlobstoreService inputService) {
-      blobstoreService = inputService; 
+   * Allow the servlet's Datastore Utterance Service to be set for mocking purposes.
+   */
+  public void setService(DatastoreUtteranceService inputService) {
```

Comment:
Should use the interface type `UtteranceService` here. If the interface alone does not have all the methods needed, and the concrete type must be used, that implies the interface is not descriptive enough and should be extended to include the necessary methods.

Reply:
adjusted